src/core/CoolBoard.cpp

```diff
@@ -32,124 +32,128 @@
 
 void CoolBoard::begin() {
   this->config();
-  pinMode(ENABLE_I2C_PIN, OUTPUT);
-  pinMode(BOOTSTRAP_PIN, INPUT);
-  digitalWrite(ENABLE_I2C_PIN, HIGH);
-  delay(100);
+  this->wifiManager.config();
+  if (!doOta) {
+    pinMode(ENABLE_I2C_PIN, OUTPUT);
+    pinMode(BOOTSTRAP_PIN, INPUT);
+    digitalWrite(ENABLE_I2C_PIN, HIGH);
+    delay(100);
 
-  this->rtc.config();
-  this->rtc.offGrid();
-  delay(100);
+    this->rtc.config();
+    this->rtc.offGrid();
+    delay(100);
 
-  this->coolBoardSensors.config();
-  this->coolBoardSensors.begin();
-  delay(100);
+    this->coolBoardSensors.config();
+    this->coolBoardSensors.begin();
+    delay(100);
 
-  this->onBoardActor.config();
-  this->onBoardActor.begin();
-  delay(100);
+    this->onBoardActor.config();
+    this->onBoardActor.begin();
+    delay(100);
 
-  this->wifiManager.config();
-  this->wifiManager.begin();
-  delay(100);
+    this->printConf();
+    this->led.printConf();
+    this->coolBoardSensors.printConf();
+    this->onBoardActor.printConf();
+    this->rtc.printConf();
 
-  this->printConf();
-  this->led.printConf();
-  this->coolBoardSensors.printConf();
-  this->onBoardActor.printConf();
-  this->wifiManager.printConf();
-  this->rtc.printConf();
+    if (this->jetpackActive) {
+      this->jetPack.config();
+      this->jetPack.begin();
+      this->jetPack.printConf();
+      delay(100);
+    }
 
-  if (this->jetpackActive) {
-    this->jetPack.config();
-    this->jetPack.begin();
-    this->jetPack.printConf();
-    delay(100);
-  }
+    if (this->ireneActive) {
+      this->irene3000.config();
+      this->irene3000.begin();
+      this->irene3000.calibrate(this->led);
+      this->irene3000.printConf();
+      delay(100);
+    }
 
-  if (this->ireneActive) {
-    this->irene3000.config();
-    this->irene3000.begin();
-    this->irene3000.calibrate(this->led);
-    this->irene3000.printConf();
-    delay(100);
-  }
+    if (this->externalSensorsActive) {
+      this->externalSensors.config();
+      this->externalSensors.begin();
+      delay(100);
+    }
 
-  if (this->externalSensorsActive) {
-    this->externalSensors.config();
-    this->externalSensors.begin();
+    this->connect();
+    this->sendPublicIP();
     delay(100);
-  }
-
-  this->connect();
-  this->sendPublicIP();
-  delay(100);
-  this->rtc.begin();
+    this->rtc.begin();
 
-  if (!this->sleepActive) {
-    sendAllConfig();
+    if (!this->sleepActive) {
+      sendAllConfig();
+    }
+  } else {
+    SPIFFS.end();
   }
   delay(100);
 }
 
 void CoolBoard::loop() {
-  INFO_LOG("Connecting...");
-  if (!this->isConnected()) {
-    this->connect();
-    INFO_LOG("Sending public IP...");
-    this->sendPublicIP();
-  }
-  INFO_LOG("Updating RTC...");
-  this->rtc.update();
+  if (!doOta) {
```

Comment:
no, no, no, no.

We don't need to check both in `begin` and `loop` if we `doOTA`. If we `doOTA`, we don't need to loop. We just need to upgrade. Please move upgrading logic to be called only from `begin`.